server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -509,6 +517,20 @@ func (s *BuildBuddyServer) CreateGroup(ctx context.Context, req *grpb.CreateGrou
 		DeveloperOrgCreationEnabled: req.GetDeveloperOrgCreationEnabled(),
 		UseGroupOwnedExecutors:      req.GetUseGroupOwnedExecutors(),
 	}
+
+	// For groups created using an API Key allow the SAML IDP Metadata URL
+	// to be inherited if the API Key group is marked as a 'parent' group.
+	// This allows the new group to be managed using a parent group API key.
+	if u.HasCapability(akpb.ApiKey_ORG_ADMIN_CAPABILITY) && u.GetUserID() == "" {
```

Comment:
nit: the `UserID == ""` check doesn't seem necessary here? (it might make sense to eventually support ORG_ADMIN for personal API keys)

Reply:
This API is also by the web frontend where u.GetUserID() != "" and I don't want this behavior to occur if someone creates a group via the UI. Once the customer is done testing, the plan is to promote the relevant APIs to the public API where we wouldn't need to do this check anymore.